sdk/Sdk/FunctionMetadataGenerator.cs

```diff
@@ -425,6 +425,54 @@ private static void AddBindingMetadata(IList<ExpandoObject> bindingMetadata, Cus
             bindingMetadata.Add(binding);
         }
 
+        /// <summary>
+        /// Creates a map of property names and its binding property names, if specified from the attribute passed in.
+        /// </summary>
+        /// <remarks>
+        /// Example output generated from BlobTriggerAttribute.
+        ///     {
+        ///         "BlobPath" : "path"
+        ///     }
+        /// 
+        /// In BlobTriggerAttribute type, the "BlobPath" property is decorated with "MetadataBindingPropertyName" attribute
+        /// where "path" is provided as the value to be used when generating metadata binding data, as shown below.
+        ///
+        ///     [MetadataBindingPropertyName("path")]
+        ///     public string BlobPath { get; set; }
+        /// 
+        /// </remarks>
+        /// <param name="attribute">The CustomAttribute instance to use for building the map. This will be a binding attribute
+        /// used to decorate the function parameter. Ex: BlobTrigger</param>
+        /// <returns>A dictionary containing the mapping.</returns>
+        private static IDictionary<string, string> GetPropertyNameAliasMapping(CustomAttribute attribute)
+        {
+            var bindingNameAliasMap = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+            TypeDefinition typeDefinition = attribute.AttributeType.Resolve();
+
+            foreach (var prop in typeDefinition.Properties)
+            {
+                var bindingPropAttr = prop.CustomAttributes.FirstOrDefault(a =>
+                                                    a.AttributeType.FullName.Equals(Constants.BindingPropertyNameAttributeType));
+
+                // The "MetadataBindingPropertyName" attribute has only one constructor which takes the name to be used.
+                var firstArgument = bindingPropAttr?.ConstructorArguments.Single();
+                if (firstArgument?.Value is null)
+                {
+                    continue;
+                }
+
+                if (firstArgument.Value.Value is not string bindingNameArgumentValue)
+                {
+                    throw new ArgumentException(
```

Comment:
I considered using InvalidOperation ,but this exception throws when the argument is of incorrect type, so chose to use this. Feel free to suggest if we can use a better exception.